### PR TITLE
add api routes for cellect

### PR DESCRIPTION
**Pull request by camallen**
Provide cellect panoptes with api endpoints to enum the workflow to preload. Long term we can look to decouple the db connector and try api end points.

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

**Comment by camallen** [comment]
I think this is ready to get it out and see how it goes. I will add more routes (and better caching via redis) if it works well so we can decouple the database adapter for cellect

File: db/structure.sql [hunk 1]
```diff
@@ -2741,13 +2749,17 @@ CREATE INDEX users_idx_trgm_login ON users USING gin ((COALESCE((login)::text, '
 
 CREATE TRIGGER tsvectorupdate BEFORE INSERT OR UPDATE ON projects FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('tsv', 'pg_catalog.english', 'display_name');
 
+ALTER TABLE projects DISABLE TRIGGER tsvectorupdate;
```
Comment (parrish) [inline db/structure.sql]: Are these meant to be in the structure?

File: db/structure.sql [hunk 2]
```diff
@@ -2741,13 +2749,17 @@ CREATE INDEX users_idx_trgm_login ON users USING gin ((COALESCE((login)::text, '
...
+
 
 --
 -- Name: tsvectorupdate; Type: TRIGGER; Schema: public; Owner: -
 --
 
 CREATE TRIGGER tsvectorupdate BEFORE INSERT OR UPDATE ON users FOR EACH ROW EXECUTE PROCEDURE tsvector_update_trigger('tsv', 'pg_catalog.english', 'login');
 
+ALTER TABLE users DISABLE TRIGGER tsvectorupdate;
```
Comment (parrish) [inline db/structure.sql]: They seem likely to be accidental

**Comment by parrish** [comment]
Looks like a few [failing specs](https://travis-ci.org/zooniverse/Panoptes/jobs/151591903) to resolve first

**Comment by camallen** [comment]
sorry for the early call - this should be good now.

File: app/controllers/cellect_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  def all_cellect_workflow_ids
+    cache_key = "#{Rails.env}#all_cellect_workflow_ids"
+    Rails.cache.fetch(cache_key, expires_in: EXPIRY) do
+      workflow_ids_using_cellect | Workflow.using_cellect.pluck(:id)
+    end
+  end
+
+  def cellect_workflow_subject_ids
+    if workflow = cellect_workflow_from_param
+      workflow.set_member_subjects.pluck(:subject_id)
```
Comment (marten) [inline app/controllers/cellect_controller.rb]: Hmm, when cellect loads these it filters out retired subjects in the query: https://github.com/zooniverse/cellect_panoptes/blob/bdb16f0bc6a375847718afc832402b3aff29fc37/lib/cellect/server/adapters/panoptes.rb#L63

Comment (camallen) [inline app/controllers/cellect_controller.rb]: good one - i'll remove them too.

**Comment by camallen** [comment]
@marten updated - now only returning non-retired subject ids for a cellect workflow

File: app/controllers/cellect_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  def launched_workflows
+    Workflow
+      .joins(:project)
+      .where("projects.launch_approved IS TRUE")
+      .select(:id, :use_cellect)
+  end
+
+  def workflow_ids_using_cellect
+    launched_workflows.collect do |w|
```
Comment (parrish) [inline app/controllers/cellect_controller.rb]: Any reason to not move this into the query rather than iterate on them?

**Comment by parrish** [comment]
This looks good for merge.  [This comment](https://github.com/zooniverse/Panoptes/pull/1902#discussion_r74614595) is pretty minor.  Happy to get this up if everybody else is.

File: app/controllers/cellect_controller.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      w.using_cellect? ? w.id : nil
+    end.compact
+  end
+
+  def all_cellect_workflow_ids
+    cache_key = "#{Rails.env}#all_cellect_workflow_ids"
+    Rails.cache.fetch(cache_key, expires_in: EXPIRY) do
+      workflow_ids_using_cellect | Workflow.using_cellect.pluck(:id)
```
Comment (marten) [inline app/controllers/cellect_controller.rb]: Why do this `|` operator rather than just `launched_workflows.using_cellect.pluck(:id)` on this line?

**Pull request merged**